### PR TITLE
Add a new route for login from the flagship app

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: docs/auth.md [hunk 1]
```diff
@@ -164,6 +164,90 @@ Set-Cookie: ...
...
+Content-Type: application/json
+```
+
+```json
+{
+  "two_factor_token": "123123123123"
+}
+```
+
+Then, the client can retry by sending the two-factor token and code:
```
Comment (Ldoppea) [inline docs/auth.md:219]: If the entered two-factor code is wrong here we get a 401 response (like when the password is not valid) and we should retry with the same token. This is different than for `/auth/session_code` which would return a new `two_factor_token` with a 403 response.

Maybe we should document that specificity?

File: web/auth/flagship.go [hunk 2]
```diff
@@ -164,3 +172,87 @@ func confirmFlagship(c echo.Context) error {
...
+		})
+	}
+
+	if inst.HasAuthMode(instance.TwoFactorMail) {
+		if len(args.TwoFactorPasscode) == 0 || len(args.TwoFactorToken) == 0 {
+			twoFactorToken, err := lifecycle.SendTwoFactorPasscode(inst)
+			if err != nil {
+				return err
+			}
+			return c.JSON(http.StatusAccepted, echo.Map{
```
Comment (Ldoppea) [inline web/auth/flagship.go:210]: Why do we return `http.StatusAccepted` instead of `http.StatusForbidden` in `CreateSessionCode`? Should we homogenise those scenario?

**Review by Ldoppea — APPROVED** [review]
The code seems to be ok. I still have some edge cases to test, but nominal scenario work well (login with/without 2FA, login with flagship certification OK/KO etc).

File: docs/auth.md (hunk 1 above)
Comment (nono) [reply docs/auth.md:219]: I have fixed the code to return a 403 in that case.

File: web/auth/flagship.go (hunk 2 above)
Comment (nono) [reply web/auth/flagship.go]: Yep, it has been fixed to `http.StatusForbidden`.
